### ISSUE TITLE
Feature: Restrict HTTPS access to API ELBs

### ISSUE THREAD

**Issue by jpb** [kubernetes-incubator/kube-aws#544]
Currently, the [Kubernetes API is available from any source IP](https://github.com/kubernetes-incubator/kube-aws/blob/15d5d8cb810688ce1a52e662cb72c1727f868c99/core/controlplane/config/templates/stack-template.json#L800). This is not ideal for a secure, production-grade cluster.

Use case: limit Kubernetes API access to a list of trusted IP addresses, such as an office's IP range.

Proposal: add `apiEndpoints[].loadBalancer.allowedSourceCIDRs` which is a collection of CIDR ranges (default to `[0.0.0.0/0]`) to allow HTTPS traffic in the API's ELB Security Group.

(Extracted from https://github.com/kubernetes-incubator/kube-aws/issues/338#issuecomment-292961597)

**Comment by mumoshu** [comment]
Submitted #552 for this.
@jpb @c-knowles Would you mind confirming if it works for you?

**Comment by jpb** [comment]
@mumoshu I've tested #552 and it works as expected 👍  Thank you for working so quickly on this - it is greatly appreciated!

One thing to note for anyone who comes across this in the future: I had my node pool in a private subnet, so to be able to give them access I had to create a second internal ELB, and set `apiEndpoints[].loadBalancer.allowedSourceCIDRs` to the VPC range. Using internal VPC IPs (or the worker SG) as a source for a public ELBs does not work. This doesn't have anything to do with the implementation in kube-aws - I believe this is an AWS issue.
